environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EncryptionSetForResourceCreationHandler.java

```diff
@@ -0,0 +1,93 @@
+
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.apache.commons.lang3.StringUtils;
+
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationEvent;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationFailureEvent;
+import com.sequenceiq.environment.environment.service.EnvironmentService;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class EncryptionSetForResourceCreationHandler extends EventSenderAwareHandler<EnvironmentDto> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EncryptionSetForResourceCreationHandler.class);
+
+    private final EnvironmentService environmentService;
+
+    private final EntitlementService entitlementService;
+
+    private final EventBus eventBus;
+
+    protected EncryptionSetForResourceCreationHandler(EventSender eventSender, EnvironmentService environmentService, EventBus eventBus,
+            EntitlementService entitlementService) {
+        super(eventSender);
+        this.environmentService = environmentService;
+        this.eventBus = eventBus;
+        this.entitlementService = entitlementService;
+    }
+
+    @Override
+    public String selector() {
+        return CREATE_ENCRYPTION_SET_FOR_RESOURCE_EVENT.selector();
+    }
+
+    @Override
+    public void accept(Event<EnvironmentDto> environmentDtoEvent) {
+        LOGGER.debug("Accepting EncryptionSetForResourceCreation event");
+        EnvironmentDto environmentDto = environmentDtoEvent.getData();
+        try {
+            environmentService.findEnvironmentById(environmentDto.getId()).ifPresent(environment -> {
+                if (AZURE.name().equalsIgnoreCase(environmentDto.getCloudPlatform())) {
+                    String encryptionKeyUrl = Optional.ofNullable(environmentDto.getParameters())
+                            .map(paramsDto -> paramsDto.getAzureParametersDto())
+                            .map(azureParamsDto -> azureParamsDto.getAzureResourceEncryptionParametersDto())
+                            .map(azureREParamsDto -> azureREParamsDto.getEncryptionKeyUrl()).orElse(null);
+                    if (StringUtils.isNotEmpty(encryptionKeyUrl)) {
+                        if (entitlementService.isAzureDiskSSEWithCMKEnabled(environmentDto.getAccountId())) {
+                            LOGGER.info("The Disk Encryption Set is to be created for {}", environmentDto.getName());
+                            /*
+                            TO DO - create the Disk Encryption Set and save environment.
```

Comment:
this implementation is missing

Reply:
Yes, There are 2 missing parts(TO DOs). One in `EncryptionSetForResourceCreationHandler.java` and other in `EncryptionSetForResourceDeletionHandler.java`. The missing parts will create and delete DES respectively. These changes will follow up in upcoming PRs for the same JIRA.